server/src/main/java/org/opensearch/cluster/service/ClusterStateStats.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service;
+
+import org.opensearch.cluster.coordination.PersistedStateStats;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Cluster state related stats.
+ *
+ * @opensearch.internal
+ */
+public class ClusterStateStats implements Writeable, ToXContentObject {
+
+    private AtomicLong stateUpdateSuccess = new AtomicLong(0);
+    private AtomicLong stateUpdateTotalTimeInMillis = new AtomicLong(0);
+    private AtomicLong stateUpdateFailed = new AtomicLong(0);
+    private PersistedStateStats remoteStateStats = null;
+
+    public ClusterStateStats() {}
+
+    public long getStateUpdateSuccess() {
+        return stateUpdateSuccess.get();
+    }
+
+    public long getStateUpdateTotalTimeInMillis() {
+        return stateUpdateTotalTimeInMillis.get();
+    }
+
+    public long getStateUpdateFailed() {
+        return stateUpdateFailed.get();
+    }
+
+    public void stateUpdated() {
+        stateUpdateSuccess.incrementAndGet();
+    }
+
+    public void stateUpdateFailed() {
+        stateUpdateFailed.incrementAndGet();
+    }
+
+    public void stateUpdateTook(long stateUpdateTime) {
+        stateUpdateTotalTimeInMillis.addAndGet(stateUpdateTime);
+    }
+
+    public void setRemoteStateStats(PersistedStateStats remoteStateStats) {
+        this.remoteStateStats = remoteStateStats;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVLong(stateUpdateSuccess.get());
+        out.writeVLong(stateUpdateTotalTimeInMillis.get());
+        out.writeVLong(stateUpdateFailed.get());
+        if (remoteStateStats != null) {
+            out.writeBoolean(true);
```

Comment:
do we need this boolean handling?

Reply:
As per other comment, you will make it Array of PersistedStateStats, so you will not need boolean. Also you can easily add local stats later if needed.

Reply:
Boolean handling removed.